gapic/schema/metadata.py

```diff
@@ -103,8 +102,13 @@ def __str__(self) -> str:
         return '.'.join(self.parent + (self.name,))
 
     @property
-    def is_external_type(self):
-        return not self.proto_package.startswith(self.api_naming.proto_package)
+    def is_proto_plus_type(self):
```

Comment:
This is non-trivial function and will benefit from adding a docstring explaining what and why you are checking for these things. Please add.

Reply:
Done in https://github.com/googleapis/gapic-generator-python/pull/1596/commits/47a2067690019bde6e17e0025ced16325ec72692